python/lsst/meas/astrom/matchPessimisticB.py

```diff
@@ -0,0 +1,594 @@
+
+    The @link lsst.pipe.base.cmdLineTask.CmdLineTask command line task@endlink
+    interface supports a flag @c -d to import @b debug.py from your
+    @c PYTHONPATH; see @ref baseDebug for more about @b debug.py files.
+
+    The available variables in MatchPessimisticBTask are:
+    <DL>
+      <DT> @c verbose (bool)
+      <DD> If True then the matcher prints debug messages to stdout
+    </DL>
+
+    To investigate the @ref meas_astrom_MatchPessimisticB_Debug, put something
+    like
+    @code{.py}
+        import lsstDebug
+        def DebugInfo(name):
+            # N.b. lsstDebug.Info(name) would call us recursively
+            debug = lsstDebug.getInfo(name)
+            if name == "lsst.pipe.tasks.astrometry":
+                debug.verbose = True
+
+            return debug
+
+        lsstDebug.Info = DebugInfo
+    @endcode
+    into your debug.py file and run this task with the @c --debug flag.
+    """
+
+    ConfigClass = MatchPessimisticBConfig
+    _DefaultName = "matchObjectsToSources"
+
+    def __init__(self, **kwargs):
+        pipeBase.Task.__init__(self, **kwargs)
+        self.makeSubtask("sourceSelector")
+
+    def filterStars(self, refCat):
+        """Extra filtering pass; subclass if desired
+        """
+        return refCat
+
+    @pipeBase.timeMethod
+    def matchObjectsToSources(self, refCat, sourceCat, wcs, refFluxField,
+                              match_tolerance=None):
+        """!Match sources to position reference stars
+
+        @param[in] refCat  catalog of reference objects that overlap the
+        exposure; reads fields for:
+            - coord
+            - the specified flux field
+        @param[in] sourceCat  catalog of sources found on an exposure; reads
+        fields for:
+            - centroid
+            - centroid flag
+            - edge flag
+            - saturated flag
+            - aperture flux, if found, else PSF flux
+        @param[in] wcs  estimated WCS
+        @param[in] refFluxField  field of refCat to use for flux
+        @param[in] match_tolerance is a MatchTolerance class object or None.
+            This this class is used to comunicate state between AstrometryTask
+            and MatcherTask. AstrometryTask will also set the MatchTolerance
+            class variable maxMatchDist based on the scatter AstrometryTask has
+            found after fitting for the wcs.
+        @return an lsst.pipe.base.Struct with fields:
+        - matches  a list of matches, each instance of
+            lsst.afw.table.ReferenceMatch
+        - usableSourcCat  a catalog of sources potentially usable for
+            matching.
+        - match_tolerance a MatchTolerance object containing the resulting
+            state variables from the match.
+        """
+        import lsstDebug
+        debug = lsstDebug.Info(__name__)
+
+        preNumObj = len(refCat)
+        refCat = self.filterStars(refCat)
+        numRefObj = len(refCat)
+
+        if self.log:
```

Comment:
Left over from MatchOptimisticB. The full line is now removed as I do not use a star filter.